packages/smooth_app/lib/pages/product/product_page.dart

```diff
@@ -73,13 +78,35 @@ class _ProductPageState extends State<ProductPage> {
     _updateHistory(context);
   }
 
+  Future<void> _refreshIfNeeded(final ProductTimestamp productTimestamp) async {
+    final int? millisecondsSinceEpoch =
+        await productTimestamp.getTimestamp(widget.product);
```

Comment:
Would it be simpler to just take the maximum of `product.lastModified` and `DaoProduct(localDatabase).getLastUpdate(product.barcode)` instead using all that abstraction? Or do you expect the `ProductTimestamp` hierarchy be useful anywhere else?

Reply:
We could probably manage to code that without abstraction, though it's not the way I code.
I do expect the `ProductTimestamp` to be used in a different use case: the timestamp for this product on the server. So far we deal with product "staleness" - and we could also deal with the product "freshness" on the server.